src/Core/Framework/DataAbstractionLayer/Search/RequestCriteriaBuilder.php

```diff
@@ -151,7 +151,7 @@ private function fromArray(array $payload, Criteria $criteria, EntityDefinition
     {
         $searchException = new SearchRequestException();
 
-        if (isset($payload['ids'])) {
+        if (isset($payload['ids']) && !empty($payload['ids'])) {
```

Comment:
not strictly necessary but !ermpty does implicitely include the `isset` check and it also avoids undefined index notices

Reply:
That is right, I have not thought about this :-) 

But maybe it makes more sense, to modify the condition, to check if there has been provided a limit, if the id array is empty.

Reply:
@aragon999 Instead of changing this here, what about adjusting the `setLimit` call below to something like this:

```
$criteria->setLimit(empty($ids) ? 0 : null);
```

Reply:
Yes, but I think this would be the same "breaking" behavior.

Reply:
well yes it is breaking in any case but is it really breaking because if you ask me it is a bug 😄

Reply:
Yes, but apparently some people are relying on this "feature" to retrieve all entities... https://xkcd.com/1172/

Reply:
@aragon999 One idea: You could add a feature flag for this and declare the old behavior as deprecated?

then you could do something like this:

```php
// @feature-deprecated tag:v6.5.0 (flag:FEATURE_NEXT_<number>) - Behavior to return all results with an empty ID array will be removed.
$criteria->setLimit(Feature::isActive('FEATURE_NEXT_12553') && empty($ids) ? 0 : null);
```

maybe you need to create a seperate `if` for the feature flag though to it can be easier removed.

See also: https://github.com/shopware/platform/commit/aeb098c730c7c6b343ca1c7dcc17f6af308d3b72#diff-e0d53f5ab2b7f5d42fc04e96e051e0c60c80e77a63fcb806ea4b83b13163912e

Reply:
@AndreasA you are right, I will implement this with a feature flag :-) Thanks a lot for the discussion and the suggestions, after that at least the interface should be consistent, and there should be no need to deprecate an empty array, I would think.